### PR TITLE
Incorrect Char array type number for portable serialization

### PR DESCRIPTION
**Pull request by ihsandemir**
Corrected the Type number at FieldType class to be the same as Java. Added read/writeBooleanArray to portable reader/writer.

Fixes https://github.com/hazelcast/hazelcast-cpp-client/issues/230

Fixes https://github.com/hazelcast/hazelcast-cpp-client/issues/236

### CODE REVIEW

**Comment by ihsandemir** [comment]
verify

File: hazelcast/include/hazelcast/client/serialization/FieldType.h [hunk 1]
```diff
@@ -53,12 +53,14 @@ namespace hazelcast {
                 static const FieldType TYPE_UTF(9);
                 static const FieldType TYPE_PORTABLE_ARRAY(10);
                 static const FieldType TYPE_BYTE_ARRAY(11);
-                static const FieldType TYPE_CHAR_ARRAY(12);
-                static const FieldType TYPE_SHORT_ARRAY(13);
-                static const FieldType TYPE_INT_ARRAY(14);
-                static const FieldType TYPE_LONG_ARRAY(15);
-                static const FieldType TYPE_FLOAT_ARRAY(16);
-                static const FieldType TYPE_DOUBLE_ARRAY(17);
+                static const FieldType TYPE_BOOLEAN_ARRAY(11);
```
Comment (sancar) [inline hazelcast/include/hazelcast/client/serialization/FieldType.h]: 12 ?

Comment (ihsandemir) [inline hazelcast/include/hazelcast/client/serialization/FieldType.h]: Ok, corrected.

**Comment by ihsandemir** [comment]
verify

**Comment by ihsandemir** [comment]
verify

**Comment by ihsandemir** [comment]
verify

**Pull request merged**